lib/fastly/client.rb

```diff
@@ -57,6 +57,8 @@ def authed?
 
     # Some methods require full username and password rather than just auth token
     def fully_authed?
+      warn("DEPRECATION WARNING: Using username/password authentication is deprecated and will not be available, 
+      please migrate to API tokens as soon as possible.")
```

Comment:
This method is called often and will end up littering customer logs with noisy output. I think we should move this warning into the `initialize` method and emit it when either `user` or `password` are provided.